cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql

```diff
@@ -211,7 +220,11 @@ class LoopWithAlloca extends Stmt {
    */
   private int getAControllingVarInitialValue(Variable var, DataFlow::Node source) {
     source = this.getAPrecedingDef(var) and
-    result = source.asExpr().getValue().toInt()
+    (
+      result = getExpr(source).(Expr).getValue().toInt()
```

Comment:
This cast isn't necessary, is it?
```suggestion
      result = getExpr(source).getValue().toInt()
```

Reply:
You're right. That seems a leftover of me trying out several things to understand what is going on.